src/shogun/multiclass/BruteKNNSolver.h

```diff
@@ -0,0 +1,59 @@
+/* This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Copyright (c) 2012-2013 Sergey Lisitsyn
+ */
+
+#ifndef BRUTESOLVER_H__
+#define BRUTESOLVER_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/lib/common.h>
+#include <shogun/multiclass/KNNSolver.h>
+#include <shogun/distance/Distance.h>
+
+namespace shogun
+{
+
+/**
+ * The BruteKNNSolver class is a solver class which inherit from KNNSolver class. It will use brute way
+ * to classify the object which mean compare the object against all training objects for each prediction.
+ */
+class CBruteKNNSolver : public CKNNSolver
+{
+	public:
+		/** default constructor */
+		CBruteKNNSolver() { init(); }
+
+		/** deconstructor */
+		~CBruteKNNSolver() { /* nothing to do */ }
+
+		/** constructor
+		 *
+		 * @param k k
+		 * @param q m_q
+		 * @param num_classes m_num_classes
+		 * @param min_label m_min_label
+		 * @param train_labels m_train_labels
+		 * @param NN nn
+		 */
+		CBruteKNNSolver(const int32_t k, const float64_t q, const int32_t num_classes, const int32_t min_label, const SGVector<int32_t> train_labels, const SGMatrix<index_t> NN);
+
+		CMulticlassLabels* classify_objects(CDistance* d, const int32_t num_lab, int32_t* train_lab, float64_t* classes);
+
+		int32_t* classify_objects_k(CDistance* d, const int32_t num_lab, int32_t* train_lab, int32_t* classes);
```

Comment:
Hi vigsterkr, maybe I haven't get the point here, but why we need to define it as virtual? There is no other class inherit from `BruteKNNSolver` and redefine `classify_objects_k`.